golem_messages/datastructures.py

```diff
@@ -68,13 +68,35 @@ def __setitem__(self, key, value):
         validator = getattr(
             self, 'validate_{}'.format(key), None
         )
-
         if callable(validator):
             validator(value=value)  # noqa pylint:disable=not-callable
 
         super().__setitem__(key, value)
 
 
+class ValidatingDependentDictItems(ValidatingDict):
```

Comment:
@kbeker please add a docstring explaining the purpose of this class, etc

Reply:
@kbeker but anyway, after reading the code more I, honestly, don't understand the real purpose of this class and I would lean towards not using it at all ... 

the only real usage in the code (apart from the very artificial usage in the test) is ensuring that `meta_parameters` are compatible with a particular `task_type`  ... which could, honestly, be achieved by just adding an explicit call to `validate_task_type_with_meta_parameters` validator to the `__init__` of the particular dict class (`ComputeTaskDef` in this case) that would explicitly use `VALID_TASK_TYPE_PAIRS` (which, btw, I'd name e.g. `TASK_TYPE_META_PARAMETERS` so that it's completely obvious what that refers to without the need to be aware of the way both `ValidatingDependentDictItems` and the validator itself work) ... and I'd make `TASK_TYPE_META_PARAMETERS` a dict, not a list of tuples since it's more natural to specify such mappings imvho...

Reply:
>which could, honestly, be achieved by just adding an explicit call to validate_task_type_with_meta_parameters validator to the __init__ of the particular dict class

Actually this class is made because I didn't want to divide validation to different places. I saw that you made `ValidatingDict` to validate messages when they are initiating so I thought that the best way to validate `meta_parameters` and `task_type` would be in another class which will do the same thing but with multiple parameters. If you want I can change it, but individually I think that it should be in separate class like `ValidatingDependentDictItems` than in the message class

>VALID_TASK_TYPE_PAIRS (which, btw, I'd name e.g. TASK_TYPE_META_PARAMETERS

Names are of course to discuss, this was just a propose and it can be changed :) 

>and I'd make TASK_TYPE_META_PARAMETERS a dict, not a list of tuples since it's more natural to specify such mappings imvho...

Also I will change it

Reply:
This has been changed. Currently this class does not exist